### ISSUE TITLE
Issues with summary writers

### ISSUE THREAD

**Issue by baggepinnen** [malmaud/TensorFlow.jl#16]
I'm having issues with the `summary_write` See the attached code for an example
```julia
using TensorFlow
using Distributions
include(Pkg.dir("TensorFlow","examples","mnist_loader.jl"))
loader = DataLoader()
session = Session(Graph())
function weight_variable(shape)
    initial = map(Float32, rand(Normal(0, .001), shape...))
    return Variable(initial)
end
function bias_variable(shape)
    initial = fill(Float32(.1), shape...)
    return Variable(initial)
end
function conv2d(x, W)
    nn.conv2d(x, W, [1, 1, 1, 1], "SAME")
end
function max_pool_2x2(x)
    nn.max_pool(x, [1, 2, 2, 1], [1, 2, 2, 1], "SAME")
end
x = placeholder(Float32)
y_ = placeholder(Float32)
W_conv1 = weight_variable([5, 5, 1, 32])
b_conv1 = bias_variable([32])
x_image = reshape(x, [-1, 28, 28, 1])
h_conv1 = nn.relu(conv2d(x_image, W_conv1) + b_conv1)
h_pool1 = max_pool_2x2(h_conv1)
W_conv2 = weight_variable([5, 5, 32, 64])
b_conv2 = bias_variable([64])
h_conv2 = nn.relu(conv2d(h_pool1, W_conv2) + b_conv2)
h_pool2 = max_pool_2x2(h_conv2)
W_fc1 = weight_variable([7*7*64, 1024])
b_fc1 = bias_variable([1024])
h_pool2_flat = reshape(h_pool2, [-1, 7*7*64])
h_fc1 = nn.relu(h_pool2_flat * W_fc1 + b_fc1)
keep_prob = placeholder(Float32)
h_fc1_drop = nn.dropout(h_fc1, keep_prob)
W_fc2 = weight_variable([1024, 10])
b_fc2 = bias_variable([10])
y_conv = nn.softmax(h_fc1_drop * W_fc2 + b_fc2)
cross_entropy = reduce_mean(-reduce_sum(y_.*log(y_conv), reduction_indices=[2]))
train_step = train.minimize(train.AdamOptimizer(1e-4), cross_entropy)
correct_prediction = indmax(y_conv, 2) .== indmax(y_, 2)
accuracy = reduce_mean(cast(correct_prediction, Float32)) # an operation
run(session, initialize_all_variables())
saver = train.Saver()
checkpoint_path = mktempdir()

weight_summary = histogram_summary("Parameters", W_fc2)
summary_writer = train.SummaryWriter("checkpoint_path")

function printtime(t0)
    dt = time()-t0
    "$(Int(dt÷60)):$(round(Int,dt % 60))"
end


    for epoch in 1:500 # 1000 epochs
        batch = next_batch(loader, 50)
        if epoch%50 == 0
            train_accuracy = run(session, accuracy, Dict(x=>batch[1], y_=>batch[2], keep_prob=>1.0))
            info("step $epoch, training accuracy $train_accuracy")
            train.save(saver, session, joinpath(checkpoint_path, "mnist"), global_step=epoch)
        end
        run(session, train_step, Dict(x=>batch[1], y_=>batch[2], keep_prob=>.5))
        summaries = run(session, weight_summary)
        write(summary_writer, summaries, global_step = epoch)
    end

    testx, testy = load_test_set()
    test_accuracy = run(session, accuracy, Dict(x=>testx, y_=>testy, keep_prob=>1.0))
    info("test accuracy $test_accuracy")
```
The errors produced are
```
julia> summaries = run(session, weight_summary)
"\nError showing value of type String:
ERROR: UnicodeError: invalid character index
 in next at ./strings/string.jl:92 [inlined]
 in escape_string(::IOContext{Base.Terminals.TTYTerminal}, ::String, ::String) at ./strings/io.jl:140
 in print_quoted at ./strings/io.jl:158 [inlined]
 in show at ./strings/io.jl:72 [inlined]
 in show at ./replutil.jl:4 [inlined]
 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::MIME{Symbol("text/plain")}, ::String) at ./REPL.jl:132
 in display(::Base.REPL.REPLDisplay{Base.REPL.LineEditREPL}, ::String) at ./REPL.jl:135
 in display(::String) at ./multimedia.jl:143
 in print_response(::Base.Terminals.TTYTerminal, ::Any, ::Void, ::Bool, ::Bool, ::Void) at ./REPL.jl:154
 in print_response(::Base.REPL.LineEditREPL, ::Any, ::Void, ::Bool, ::Bool) at ./REPL.jl:139
 in (::Base.REPL.##22#23{Bool,Base.REPL.##33#42{Base.REPL.LineEditREPL,Base.REPL.REPLHistoryProvider},Base.REPL.LineEditREPL,Base.LineEdit.Prompt})(::Base.LineEdit.MIState, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Bool) at ./REPL.jl:652
 in run_repl(::Base.REPL.LineEditREPL, ::Base.##930#931) at ./REPL.jl:188
```
and
```
ERROR: MethodError: no method matching write(::TensorFlow.train.SummaryWriter, ::String; global_step=1)
Closest candidates are:
  write(::TensorFlow.train.SummaryWriter, ::String) at /local/home/fredrikb/.julia/v0.5/TensorFlow/src/summary_writer.jl:46 got unsupported keyword argument "global_step"
```

My system info
```
julia> versioninfo()
Julia Version 0.5.0-rc3+0
Commit e6f843b* (2016-08-22 23:43 UTC)
Platform Info:
  System: Linux (x86_64-redhat-linux)
  CPU: Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz
  WORD_SIZE: 64
  BLAS: libopenblas (DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblasp.so.0
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, haswell)
```

**Comment by malmaud** [comment]
The first error isn't really significant: The REPL can't display the summary object because it is not a valid unicode string, but that won't effect the ability to save it. 

The second error is because the documentation is wrong: Just write `write(summary_writer, summaries, epoch)`. 

I'll fix both these issues and thank you for reporting them!

**Comment by malmaud** [comment]
Closed in 9317fe827fdd9ef2d6b8eb82611d9af683853e34
